Add toggles for disabling heart drops and random drops

1SDANi opened the pull request:
No Random Drops: Disables random drops, except from the Goron Pot, Dampe, and bosses
No Heart Drops: Disables heart drops, but not heart placements, like from a Deku Scrub

1SDANi committed:
Adds toggles for disabling random drops and fixed heart drops

1SDANi committed:
Updated to be compatible with the latest develop

1SDANi committed:
Merge remote-tracking branch 'remotes/personal/hero_toggles' into hero_toggles

1SDANi committed:
Changed the "No Fixed Heart Drops" toggle to the more useful "No Random Heart Drops" toggle

1SDANi committed:
Updated to latest develop

Kenix3 commented on libultraship/libultraship/SohImGuiImpl.cpp:
```diff
@@ -915,6 +915,10 @@ namespace SohImGui {
+                    EnhancementCheckbox("No Random Drops", "gNoRandomDrops");
+                    Tooltip("Disables random drops, except from the Goron Pot, Dampe, and bosses");
+                    EnhancementCheckbox("No Heart Drops", "gNoHeartDrops");
+                    Tooltip("Disables heart drops, but not heart placements, like from a Deku Scrub running off");
```
```suggestion
                    Tooltip("Disables heart drops, but not heart placements, like from a Deku Scrub running off. This simulates Hero Mode from other games in the series.");
```

1SDANi replied:
I mean, it simulates the "no heart drops" aspect of Hero Mode, not Hero Mode in its entirety, but sure. I'll make that change regardless.

1SDANi committed:
Adds an explanation that the no heart drops option is similar to Hero Mode

1SDANi committed:
Merge branch 'develop' of https://github.com/HarbourMasters/Shipwright into hero_toggles

1SDANi committed:
Moved difficulty options into their own sub-menu
Moved time saver options into their own sub-menu
Moved clutter reducing options into their own sub-menu

1SDANi commented:
One change I forgot to mention in the latest commit, I made the default text speed 2 as discussed on discord

louist103 commented on libultraship/libultraship/SohImGuiImpl.cpp:
```diff
@@ -876,43 +876,67 @@ namespace SohImGui {
+                    if (ImGui::BeginMenu("Time Savers"))
+                    {
+                        EnhancementSliderInt("Text Speed: %dx", "##TEXTSPEED", "gTextSpeed", 1, 5, "");
+                        EnhancementSliderInt("King Zora Speed: %dx", "##WEEPSPEED", "gMweepSpeed", 1, 5, "");
```
```suggestion
                        EnhancementSliderInt("King Zora Speed: %dx", "##MWEEPSPEED", "gMweepSpeed", 1, 5, "");
```

1SDANi replied:
done

1SDANi committed:
fixed the tag name of mweep speed

louist103 commented:
Merge pending rebase conflicts.

vaguerant commented:
Sorry to make a suggestion on a PR that looks ready to merge, but since this is already streamlining the option sorting, this seems like a good place to implement this, as discussed on Discord.
> **Kenix** June 23, 2022 3:12 AM
>The right now answer is that skip text is moved to the experimental features. A PR doing this would be accepted.

https://discord.com/channels/808039310850130000/955997420288888902/989216233331261450

The Skip Text feature is known to cause at least one softlock, and Kenix is on-board to move it to the Experimental section while that is the case.

1SDANi committed:
Moved Skip Text to Experimental

1SDANi commented:
Done, also gave a warning about the water temple softlock and how to fix it.

1SDANi committed:
Merge branch 'develop' of https://github.com/HarbourMasters/Shipwright into hero_toggles

1SDANi commented:
The merge I just made and pushed was because prior to it the fact that I made changes to SohImGuiImpl.cpp whilst a recently merged PR renamed that file meant that github was thinking there was a conflict when there wasn't. Merging develop onto my branch told github that the changes to that file are meant to apply to the identical file with a slightly different name.

jbodner09 commented:
> Done, also gave a warning about the water temple softlock and how to fix it.

Just want to confirm that the "fix" still works after #454 is in the builds.  The issue where D-pad ocarina notes causes the camera to reset outside of debug mode (#448) was caused by this debug command not being guarded by a cvar, and the cvar I used to fix the issue was `gDebugCamera` since that was what was used elsewhere in that file, which is not the main debug mode cvar.

1SDANi commented:
It may not be the main debug mode cvar, but from some light testing, it has become apparent that gDebugCamera is toggled by the checkbox that toggles the main debug mode cvar. Try it yourself, you cannot use camera controls whilst Debug Mode is disabled, but you can whilst it is enabled.
